addons/calico/3.18.1/install.sh

```diff
@@ -0,0 +1,99 @@
+
+CALICO_DISABLE_ENCRYPTION=0 # setting from yaml spec
+CALICO_WIREGUARD=0 # may be false in dev workflow, even when encryption is enabled
+
+function calico_pre_init() {
+    EXISTING_POD_CIDR=$(kubectl -n kube-system get daemonset calico-node -ojsonpath='{ .spec.template.spec.containers[0].env[?(@.name=="CALICO_IPV4POOL_CIDR")].value}' 2>/dev/null)
+    echo "FOUND EXISTING POD CIDR $EXISTING_POD_CIDR"
+}
+
+function calico() {
+    local src="$DIR/addons/calico/$CALICO_VERSION"
+    local dst="$DIR/kustomize/calico"
+
+    # Cannot migrate from weave to calico
+    if calico_weave_conflict; then
+        printf "${YELLOW}Cannot migrate from weave to calico${NC}\n"
+        return 0
+    fi
+
+    # can't use kustomize because the CRDs must be applied first
+    cp "$src/calico.yaml" "$dst/"
+    cp "$src/kustomization.yaml" "$dst/"
+    cp "$src/ip-in-ip-encapsulation.yaml" "$dst/"
+    render_yaml_file "$src/tmpl-daemonset-pod-cidr.yaml" > "$dst/daemonset-pod-cidr.yaml"
+    kubectl apply -k "$dst/"
+
+    calico_cli
+    calico_wireguard
+
+    echo "Waiting for default FelixConfiguration to configure encryption"
+    if ! spinner_until 180 kubernetes_cluster_resource_exists felixconfiguration default; then
+        bail "Failed to find default FelixConfiguration resource"
+    fi
+    if [ "$CALICO_WIREGUARD" = "1" ]; then
+        calicoctl patch felixconfiguration default --type='merge' -p '{"spec":{"wireguardEnabled":true}}'
+    else
+        calicoctl patch felixconfiguration default --type='merge' -p '{"spec":{"wireguardEnabled":false}}'
+    fi
+}
+
+function calico_join() {
+    # Cannot migrate from weave to calico
+    if calico_weave_conflict; then
+        printf "${YELLOW}Cannot migrate from weave to calico${NC}\n"
+        return 0
+    fi
+
+    calico_cli
+    calico_wireguard
+}
+
+function calico_cli() {
+    local src="$DIR/addons/calico/$CALICO_VERSION"
+
+    if [ ! -f "$src/assets/calicoctl" ] && [ ! "$AIRGAP" = "1" ]; then
+        mkdir -p "$src/assets"
+        curl -L https://github.com/projectcalico/calicoctl/releases/download/v3.18.0/calicoctl > "$src/assets/calicoctl"
+    fi
+
+    chmod +x "$src/assets/calicoctl"
+    mv "$src/assets/calicoctl" /usr/local/bin/
```

Comment:
Wouldn't this only apply in the non-airgap case? I didn't see calicoctl in the manifest.